lib/route_translator/translator.rb

```diff
@@ -35,27 +36,31 @@ def translate_name(name, locale, named_routes_names)
 
     module_function
 
-    def translations_for(route_set, path, name, options_constraints, options)
-      RouteTranslator::Translator::RouteHelpers.add name, route_set.named_routes
+    def translations_for(route)
+      RouteTranslator::Translator::RouteHelpers.add route.name, route.route_set.named_routes
 
       available_locales.each do |locale|
         begin
-          translated_path = RouteTranslator::Translator::Path.translate(path, locale)
+          translated_path = RouteTranslator::Translator::Path.translate(route.path, locale, route.scope)
         rescue I18n::MissingTranslationData => e
           raise e unless RouteTranslator.config.disable_fallback
           next
         end
 
-        translated_options_constraints = options_constraints.dup
-        translated_options             = options.dup
+        yield translated_attributes(locale, translated_path, route)
+      end
+    end
 
-        translated_options_constraints[RouteTranslator.locale_param_key] = locale.to_s
-        translated_options[RouteTranslator.locale_param_key]             = locale.to_s.gsub('native_', '') unless translated_options.include?(RouteTranslator.locale_param_key)
+    def translated_attributes(locale, translated_path, route)
```

Comment:
`translated_attributes` is not used outside this module, so you should move it inside `class << self`